File: AtCoder/ABC026/d.c

#include <stdio.h>
#include <math.h>
#include <float.h>

int main(void){

  int a,b,c;
  scanf("%d %d %d",&a,&b,&c);
  double bgn = 0.0;
  double end = 2000.0;

  while(1){
    double t = 0.5 * (bgn+end);
    if ( 100.0 < a*t + b*sin(c*t*M_PI) ){
      end = t;
    }else{
      bgn = t;
    }
    if ( end-bgn < 0.000000000001 )break;
  }
  printf("%.10f\n",bgn);

}

File: AtCoder/ABC026/d.java
import java.util.Scanner;
import java.util.stream.*;
import java.util.Arrays;
import java.util.function.Function;
  
public class Main {

  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    final int a = sc.nextInt();
    final int b = sc.nextInt();
    final int c = sc.nextInt();

    double l = 0d;
    double r = 200d;

    Function<double,double> f = ( p ) -> {
      return (double)a*p + (double)b*Math.sin( (double)c * p * Math.PI );
    };

    //while( l-r < 1.0e-){
      //if 
    //}

    System.out.println( l );
  }
  
}